### ISSUE TITLE
As a SO I would like to choose how my site search form works

### ISSUE THREAD

**Issue by kevincrafts** [CuBoulder/express#1108]
## Context

Depending on the site, the search form could have some configuration options:

1. Search this site (the current option)
2. Search all of CU Boulder
3. Search a collection of sites (maybe at a later date?)

Site visitors should see either:

a search box that only has a single option (1 or 2 above: search label should be "search this site" or "search cu boulder/search colorado.edu") 

OR

a search box with an multiple options to search (1 and 2 and maybe 3 above)

![screen shot 2017-06-01 at 11 47 53 am](https://cloud.githubusercontent.com/assets/349713/26693098/38c38fde-46c0-11e7-8a28-ae956087a8fe.jpg)

Being able to define a number of sites as a collection could help users searching for information. Users may not know that there is another site with the information they look for. Example: College of Engineering and the various subsites we created. Should each of of the sites search all of the college of engineering collection, or be given the option to search that site, vs, the collection of sites, vs all of campus.

**Comment by jwfuller** [comment]
Currently we define a single collection per site. 

Maintaining this would not be a trivial task, we would likely need to write a new endpoint for Atlas

**Comment by alexfinnarn** [comment]
Keep in mind the move to Google CSE with this epic. IT won't be long before that switch would happen. While the GSA has single collections per site, Google CSE is very basic. Without using the API, which has a fee for our purposes, to have a site collection you would need to set it up like Google Analytics with your own CSE ID. 

There are rules for inclusion/exclusion (and many others), but they can't be dynamic and so adding a "search your site" radio should be progressively added only when they have created an account for that purpose. 

The "collection of sites" would be easier done this way as they have to create them, but it shouldn't be hard to explain how to do that. Not sure if development time on Atlas should go into that feature as it wouldn't be used after GSA deprecated.  

If people are to use an external Google CSE service that they can control, I'd say the security tag should go back on this epic. Only allowing a search for the whole domain would allow us to have full control, but allowing them to have their own account means we have to make sure they can't sabotage anything.

For testing purposes, we have no tests related to search, but switching to CSE brings in the question of how to test external services. Since we have no current search tests and aren't likely to add them, I will remove the tag and link to some "how to do external testing" issue.

**Comment by kevincrafts** [comment]
![screen shot 2017-08-04 at 11 00 08 am](https://user-images.githubusercontent.com/349713/28978678-1f84cc06-7904-11e7-9ec9-a0d222d45e8a.jpg)

**Comment by kevincrafts** [comment]
This site option is only avail be if there is a CSE ID.
